app/services/NextUpdatesService.scala

```diff
@@ -100,4 +103,43 @@ class NextUpdatesService @Inject()(val incomeTaxViewChangeConnector: IncomeTaxVi
       }
     }
   }
+
+
+  def getObligationDates(id: String)
+                        (implicit user: MtdItUser[_], ec: ExecutionContext, hc: HeaderCarrier): Future[Seq[DatesModel]] = {
+    getNextUpdates() map {
+      case NextUpdatesErrorModel(code, message) =>
+        Logger("application").error(
+          s"[BusinessAddedObligationsController][handleRequest] - Error: $message, code $code")
+        Seq.empty
+      case NextUpdateModel(start, end, due, _, _, periodKey) =>
+        Seq(DatesModel(start, end, due, periodKey, isFinalDec = false))
+      case model: ObligationsModel =>
+        Seq(model.obligations.flatMap(x => x.currentCrystDeadlines) map {
+          source =>
+            DatesModel(source.start, source.end, source.due, source.periodKey, isFinalDec = true)
+        },
+          model.obligations.filter(x => x.identification == id).flatMap(obligation => obligation.obligations.map(x => DatesModel(x.start, x.end, x.due, x.periodKey, isFinalDec = false)))
+        ).flatten
+    }
+  }
+
+  def getObligationsViewModel(id: String, showPreviousTaxYears: Boolean)
+                             (implicit user: MtdItUser[_], ec: ExecutionContext, hc: HeaderCarrier): Future[ObligationsViewModel] = {
+    val processingRes = for {
```

Comment:
there are chances that this transformation can failed:
* filtering on the line 134
* use of .head on line 135

Reply:
I thought we agreed the .head was fine because x.partition(y => y==x.head) just returns two empty lists if x is empty, and .filter will be the same thing?